src/System.Private.ServiceModel/src/Internals/System/Xml/XmlMtomWriter.cs

```diff
@@ -0,0 +1,1553 @@
+        }
+
+        private XmlDictionaryWriter Writer
+        {
+            get
+            {
+                if (!IsInitialized)
+                {
+                    Initialize();
+                }
+                return _writer;
+            }
+        }
+
+        private bool IsInitialized
+        {
+            get { return (_initialContentTypeForRootPart == null); }
+        }
+
+        private void Initialize(Stream stream, string startInfo, string boundary, string startUri, bool writeMessageHeaders, bool ownsStream)
+        {
+            if (stream == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(stream));
+            if (startInfo == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(startInfo));
+            if (boundary == null)
+                boundary = GetBoundaryString();
+            if (startUri == null)
+                startUri = GenerateUriForMimePart(0);
+            if (!MailBnfHelper.IsValidMimeBoundary(boundary))
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentException(SR.Format(SR.MtomBoundaryInvalid, boundary), nameof(boundary)));
+
+            _ownsStream = ownsStream;
+            _isClosed = false;
+            _depth = 0;
+            _totalSizeOfMimeParts = 0;
+            _sizeOfBufferedBinaryData = 0;
+            _binaryDataChunks = null;
+            _contentType = null;
+            _contentTypeStream = null;
+            _contentID = startUri;
+            if (_mimeParts != null)
+                _mimeParts.Clear();
+            _mimeWriter = new MimeWriter(stream, boundary);
+            _initialContentTypeForRootPart = GetContentTypeForRootMimePart(_encoding, startInfo);
+            if (writeMessageHeaders)
+                _initialContentTypeForMimeMessage = GetContentTypeForMimeMessage(boundary, startUri, startInfo);
+        }
+
+        private void Initialize()
+        {
+            if (_isClosed)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.XmlWriterClosed));
+
+            if (_initialContentTypeForRootPart != null)
+            {
+                if (_initialContentTypeForMimeMessage != null)
+                {
+                    _mimeWriter.StartPreface();
+                    _mimeWriter.WriteHeader(MimeGlobals.MimeVersionHeader, MimeGlobals.DefaultVersion);
+                    _mimeWriter.WriteHeader(MimeGlobals.ContentTypeHeader, _initialContentTypeForMimeMessage);
+                    _initialContentTypeForMimeMessage = null;
+                }
+
+                WriteMimeHeaders(_contentID, _initialContentTypeForRootPart, _isUTF8 ? MimeGlobals.Encoding8bit : MimeGlobals.EncodingBinary);
+
+                Stream infosetContentStream = _mimeWriter.GetContentStreamAsync().GetAwaiter().GetResult();
```

Comment:
Why not just call the sync version of GetContentStream()? You haven't mixed sync/async in any other places.

Reply:
I originally made everything async, then hit this code so brought back the sync version to use here. Looks like I forgot to actually use it here. I'll fix this.